pkg/fleetautoscalers/controller_test.go

```diff
@@ -695,6 +720,92 @@ func TestControllerUpdateStatusUnableToScale(t *testing.T) {
 	})
 }
 
+func TestControllerSyncFleetAutoscalerWithCustomSyncInterval(t *testing.T) {
+	t.Run("create fas thread", func(t *testing.T) {
+		t.Parallel()
+		c, m := newFakeController()
+		fc := clock.NewFakeClock(time.Now())
+		c.clock = fc
+
+		fas, f := defaultFixtures()
+		fasKey := fas.Namespace + "/" + fas.Name
+		fas.Spec.Sync.FixedInterval.Seconds = 10
+
+		fasUpdatedCount := counter{v: 0}
+
+		m.AgonesClient.AddReactor("list", "fleetautoscalers", func(action k8stesting.Action) (bool, runtime.Object, error) {
+			return true, &autoscalingv1.FleetAutoscalerList{Items: []autoscalingv1.FleetAutoscaler{*fas}}, nil
+		})
+
+		m.AgonesClient.AddReactor("update", "fleetautoscalers", func(action k8stesting.Action) (bool, runtime.Object, error) {
+			ca := action.(k8stesting.UpdateAction)
+			fas := ca.GetObject().(*autoscalingv1.FleetAutoscaler)
+			fasUpdatedCount.Inc()
+			return true, fas, nil
+		})
+
+		m.AgonesClient.AddReactor("list", "fleets", func(action k8stesting.Action) (bool, runtime.Object, error) {
+			return true, &agonesv1.FleetList{Items: []agonesv1.Fleet{*f}}, nil
+		})
+
+		ctx, cancel := agtesting.StartInformers(m, c.fleetSynced, c.fleetAutoscalerSynced)
+		defer cancel()
+
+		err := c.syncFleetAutoscalerWithCustomSyncInterval(ctx, fasKey)
+		assert.Nil(t, err)
+		assert.Contains(t, c.fasThreads, fasKey)
+		// set the clock forward by one sync interval forward, the fas update function should be called twice
+		c.clock.Sleep(time.Duration(fas.Spec.Sync.FixedInterval.Seconds) * time.Second)
+		// we need a small block here so that autoscale rountine can run
+		time.Sleep(1 * time.Second)
+		assert.Equal(t, uint32(2), fasUpdatedCount.Value())
```

Comment:
This is looking so much better 🥳 . Rather than a single sleep, which could be flaky, let's use an:
https://pkg.go.dev/github.com/stretchr/testify@v1.7.0/assert#Eventually

(let's see if I can get this right doing this within Github)

```suggestion
		assert.Eventually(t, func() bool {
		   return uint32(2) == fasUpdatedCount.Value()
		}, 10 * time.Second, 500 * time.Milliseconds)
```

Then if the processing takes a little longer than expected, we can retry a few times.